drivers/storage/portworx/deployment.go

```diff
@@ -796,16 +796,24 @@ func (t *template) getEnvList() []v1.EnvVar {
 	}
 
 	if t.cluster.Spec.ImagePullSecret != nil && *t.cluster.Spec.ImagePullSecret != "" {
-		envMap["REGISTRY_CONFIG"] = &v1.EnvVar{
-			Name: "REGISTRY_CONFIG",
-			ValueFrom: &v1.EnvVarSource{
-				SecretKeyRef: &v1.SecretKeySelector{
-					Key: ".dockerconfigjson",
-					LocalObjectReference: v1.LocalObjectReference{
-						Name: *t.cluster.Spec.ImagePullSecret,
+		pxVer2_3_2, _ := version.NewVersion("2.3.2")
+		if t.pxVersion.LessThan(pxVer2_3_2) {
+			envMap["REGISTRY_CONFIG"] = &v1.EnvVar{
+				Name: "REGISTRY_CONFIG",
+				ValueFrom: &v1.EnvVarSource{
+					SecretKeyRef: &v1.SecretKeySelector{
+						Key: ".dockerconfigjson",
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: *t.cluster.Spec.ImagePullSecret,
+						},
 					},
 				},
-			},
+			}
+		} else {
+			envMap["REGISTRY_SECRET"] = &v1.EnvVar{
+				Name:  "REGISTRY_SECRET",
+				Value: fmt.Sprintf("%s/%s", t.cluster.Namespace, *t.cluster.Spec.ImagePullSecret),
```

Comment:
Note about namespace here:
* Kubernetes own `imagePullSecrets` directive does not normally allow Pull-Secrets to span between the namespaces (i.e. one can use only "local namespace" secrets)
* our implementation of `REGISTRY_SECRET` is a bit different, since it'll allow fetching secrets also from a different namespace  (note, we needed this to support specific IBM-K8s-cloud default setups)
* the newer versions of OCI-Monitor's do not need the `REGISTRY_SECRET` setting at all -- by default they'll try to reuse the pull-secret that's been assigned to OCI-Monitor POD itself
  - so, in the future PX-Operator releases we'll likely drop the `REGISTRY_SECRET` env. variable altogether

Because of this, it might be better to align with K8s limitation (and our future default), and not use Namespace in `REGISTRY_SECRET`  (i.e. use `<pullSecret>` notation, instead of our custom `<namespace>/<pullSecret>`)

Reply:
@zoxpx So you recommend not passing REGISTRY_SECRET at all? Or asking to pass only the secret name?
I have to pass the namespace because from the ticket it seemed like oci-mon defaults to kube-system if namespace is not passed. Even operator mandates user to have pull secret in the same namespace. But if cluster itself if deployed in non-kube-system ns, I need to pass the namespace for oci-mon.

Reply:
If namespace is not specified in `REGISTRY_SECRET`, the OCI-Monitor will use its own namespace to access the secrets.
Apropos removing the `REGISTRY_SECRET` variable altogether, I think it's too early for this  (we've introduced this default fairly recently, if I remember correctly)

Reply:
@zoxpx has this always been the case? Can I safely assume even for 2.3.2 we don't need to pass the namespace in REGISTRY_SECRET?

Reply:
yes and yes  :)